cli/cmd/plugin/unmanaged-cluster/cluster/kind.go

```diff
@@ -53,11 +53,20 @@ func (kcm KindClusterManager) Create(c *config.UnmanagedClusterConfig) (*Kuberne
 	kindProvider := kindcluster.NewProvider()
 	clusterConfig := kindcluster.CreateWithKubeconfigPath(c.KubeconfigPath)
 
-	// TODO(stmcginnis): Determine what we need to do for kind configuration
-	parsedKindConfig, err := kindConfigFromClusterConfig(c)
-	if err != nil {
-		return nil, err
+	// If a user has provided something in the ProviderConfiguration,
+	// assume it is a full kind configuration and don't attempt to produce
+	// a configuration from the unmanaged-cluster config
+	var parsedKindConfig []byte
+	var err error
+	if c.ProviderConfiguration != "" {
+		parsedKindConfig = []byte(c.ProviderConfiguration)
+	} else {
+		parsedKindConfig, err = kindConfigFromClusterConfig(c)
+		if err != nil {
+			return nil, err
```

Comment:
We need to improve this error message when the provider configuration is returned.

It's good to return the error for `kind`, but the user needs context that it failed to evaluate the provide config.

![image](https://user-images.githubusercontent.com/6200057/156801402-ab487e8b-ce97-4886-9f12-feb59ac487ad.png)

Reply:
Looking closer at the code, I now see there is no pre-config validation before running the cluster create. So my recommendation is to improve the error as described in https://github.com/vmware-tanzu/community-edition/pull/3304/files#r819755863.

Then, simply mention that `kind` returned the error. So my suggestion would be:

```go
kindConfig := kindcluster.CreateWithRawConfig(parsedKindConfig)
err = kindProvider.Create(c.ClusterName, clusterConfig, kindConfig)
if err != nil {
    return nil, fmt.Errorf("kind returned error: %s", err)
}

```